kansha/board/comp.py

```diff
@@ -161,30 +163,55 @@ def many_user_render(h, number):
 
         self.must_reload_search = False
 
+    @classmethod
+    def get_id_by_uri(cls, uri):
+        board = DataBoard.get_by_uri(uri)
+        board_id = None
+        if board is not None:
+            board_id = board.id
+        return board_id
+
+    @classmethod
+    def exists(cls, **kw):
+        return DataBoard.exists(**kw)
+
+    def reload_save_template_comp(self):
+        self.save_template_comp = component.Component(SaveTemplateTask(self))
+        self.save_template_comp.on_answer(lambda v: self.reload_save_template_comp())
+
     def copy(self, owner, additional_data):
         new_data = self.data.copy(None)
+        if self.data.background_image:
+            new_data.background_image = self.assets_manager.copy(self.data.background_image)
         new_obj = self._services(Board, new_data.id, self.app_title, self.app_banner, self.theme, self.card_extensions, self.search_engine, load_data=False)
         new_obj.add_member(owner, 'manager')
         additional_data['author'] = owner
 
+        additional_data['labels'] = []
+        for lbl in self.labels:
+            new_label = lbl.copy(new_obj, additional_data)
+            additional_data['labels'].append(new_label)
+            new_obj.labels.append(new_label)
+
         cols = [col() for col in self.columns if not col().is_archive]
-        for index, column in enumerate(cols):
+        index = 0
+        for column in cols:
             new_col = column.copy(new_obj, additional_data)
             new_obj.columns.append(component.Component(new_col))
+            index += 1
 
-        new_obj.archive_column = new_obj.create_column(index=index + 1, title=_(u'Archive'), archive=True)
-
-        for lbl in self.labels:
-            new_obj.labels.append(lbl.copy(new_obj, additional_data))
+        new_obj.archive_column = new_obj.create_column(index=index, title=_(u'Archive'), archive=True)
```

Comment:
Why not simply new_obj.create_column(index=len(new_obj.columns), title=_(u'Archive'), archive=True)?